File: ObjectiveCSample/AdWhaleSample-ObjC/NativeAd/NativeAdView.m
//
//  NativeAdView.m
//  AdWhaleSample-ObjC
//
//  Created by FSN on 6/25/24.
//

#import "NativeAdView.h"

@implementation NativeAdView

- (UILabel * _Nonnull)adBodyLabel {
    return _bodyLabel;
}

- (UIButton * _Nonnull)adCallToActionButton {
    return _callToActionButton;
}

- (AdWhaleMediaView * _Nonnull)adMediaView {
    return _mediaView;
}

- (UIImageView * _Nonnull)adProfileIconView {
    return _profileIconView;
}

- (UILabel * _Nonnull)adProfileNameLabel {
    return _profileNameLabel;
}

- (UILabel * _Nonnull)adTitleLabel {
    return _titleLabel;
}

@end

File: SwiftSample/AdWhaleSample-Swift/NativeAdView.swift
//
//  NativeAdView.swift
//  AdWhaleSample-Swift
//
//  Created by FSN on 5/8/24.
//

import UIKit
import AdWhaleSDK

class NativeAdView: AdWhaleNativeAdView {

    @IBOutlet var titleLabel: UILabel!
    @IBOutlet var bodyLabel: UILabel!
    @IBOutlet var callToActionButton: UIButton!
    @IBOutlet var profileNameLabel: UILabel!
    @IBOutlet var profileIconView: UIImageView!
    @IBOutlet var adWhaleMediaView: AdWhaleMediaView!
    
    override func adTitleLabel() -> UILabel {
        return titleLabel
    }
    
    override func adBodyLabel() -> UILabel {
        return bodyLabel
    }
    
    override func adCallToActionButton() -> UIButton {
        return callToActionButton
    }
    
    override func adProfileNameLabel() -> UILabel {
        return profileNameLabel
    }
    
    override func adProfileIconView() -> UIImageView {
        return profileIconView
    }
    
    override func adMediaView() -> AdWhaleMediaView {
        return adWhaleMediaView
    }

}
